frontend/src/component/personalDashboard/ProjectSetupComplete.tsx

```diff
@@ -40,6 +40,91 @@ export const ProjectSetupComplete: FC<{ project: string }> = ({ project }) => {
+    insights: PersonalDashboardProjectDetailsSchemaInsights,
+) => {
+    let trend: HeathTrend = 'unknown';
+    if (
+        insights.avgHealthCurrentWindow !== null &&
+        insights.avgHealthPastWindow !== null
+    ) {
+        if (insights.avgHealthCurrentWindow > insights.avgHealthPastWindow) {
+            trend = 'improved';
+        } else if (
+            insights.avgHealthCurrentWindow < insights.avgHealthPastWindow
+        ) {
+            trend = 'declined';
+        } else if (
+            insights.avgHealthPastWindow === insights.avgHealthCurrentWindow
+        ) {
+            trend = 'consistent';
+        }
+    }
+    return trend;
+};
+
+export const ProjectSetupComplete: FC<{
+    project: string;
+    insights: PersonalDashboardProjectDetailsSchemaInsights;
+}> = ({ project, insights }) => {
+    const projectHealthTrend = determineProjectHealthTrend(insights);
+
+    return (
+        <ActionBox>
+            <TitleContainer>
+                <Lightbulb color='primary' />
+                <h3>Project Insight</h3>
+            </TitleContainer>
+
+            {projectHealthTrend === 'unknown' ? (
+                <ConnectedSdkProject project={project} />
+            ) : null}
+            {projectHealthTrend === 'improved' ? (
+                <Typography>
+                    On average, your project health went up from{' '}
+                    <PercentageScore>
+                        {insights.avgHealthPastWindow}%
+                    </PercentageScore>{' '}
+                    to{' '}
+                    <PercentageScore>
+                        {insights.avgHealthCurrentWindow}%
+                    </PercentageScore>{' '}
+                    during the last 4 weeks.
+                </Typography>
+            ) : null}
+            {projectHealthTrend === 'declined' ? (
+                <Typography>
+                    On average, your project health went down from{' '}
+                    <PercentageScore>
+                        {insights.avgHealthPastWindow}%
+                    </PercentageScore>{' '}
+                    to{' '}
+                    <PercentageScore>
+                        {insights.avgHealthCurrentWindow}%
+                    </PercentageScore>{' '}
+                    during the last 4 weeks.
+                </Typography>
+            ) : null}
+            {projectHealthTrend === 'consistent' ? (
+                <Typography>
+                    On average, your project health has remained at{' '}
+                    <PercentageScore>
+                        {insights.avgHealthCurrentWindow}%
+                    </PercentageScore>{' '}
+                    during the last 4 weeks.
+                </Typography>
+            ) : null}
+            {projectHealthTrend !== 'unknown' ? (
+                <Link to={`/projects/${project}/insights`}>
+                    View more insights
+                </Link>
+            ) : null}
```

Comment:
I'd maybe consider extracting this into a variable above instead of a series of ternaries. Just to avoid any future issues 🤷🏼

Reply:
Refactored to a separate component with early returns